dask/dataframe/tests/test_io.py

```diff
@@ -668,3 +668,29 @@ def test_report_dtype_correction_on_csvs():
             assert False
         except ValueError as e:
             assert "'numbers': 'float64'" in str(e)
+
+
+def test_hdf_globbing():
+    pytest.importorskip('tables')
+    df = pd.DataFrame({'x': ['a', 'b', 'c', 'd'],
+                       'y': [1, 2, 3, 4]}, index=[1., 2., 3., 4.])
+    a = dd.from_pandas(df, 2)
+
+    pattern = '/tmp/*foo.h5'
+    with tmpfile('one_foo.h5') as fn1:
+        a.to_hdf(fn1, '/foo/data', format='table')
```

Comment:
It might be nice to use `df.to_hdf` here so that this test is not sensitive to that code.